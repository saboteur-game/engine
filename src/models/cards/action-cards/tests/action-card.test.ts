import CardParameters from "../../card-parameters";
import ActionCard from "../action-card";

describe("ActionCard", () => {
  let actionCard: ActionCard;
  let cardParameters: CardParameters;

  beforeEach(() => {
    actionCard = new ActionCard();
    cardParameters = { position: "0,0" };
  });

  it("can be serialized", () => {
    expect(actionCard.toJS()).toMatchSnapshot();
  });

  describe("play a card", () => {
    it("returns the played card", () => {
      expect(actionCard.play(cardParameters)).toBe(actionCard);
    });
  });
});
