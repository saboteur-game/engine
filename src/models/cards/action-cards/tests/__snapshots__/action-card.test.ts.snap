// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionCard can be serialized 1`] = `
Object {
  "id": "test-id-0",
  "parameters": undefined,
  "status": "unused",
}
`;
